### PR TITLE
Add the std test suite.

### PR DESCRIPTION
**Pull request by StephanTLavavej**
# Description

Works towards #143. Note that this is just the source code; the infrastructure to automatically build and run the tests is coming soon.

NOTICE.txt
Mention the LLVM Project, as its code appears
in tests/std/tests/P0220R1_optional/test.cpp and elsewhere.

azure-devops/enforce-clang-format.cmd
Process everything within tests. This includes libcxx, std, and tr1.

docs/cgmanifest.json
Update this file for Microsoft-internal purposes. (It records the commit
hashes of repos whose source code we've incorporated into our own repo.
For llvm-project, this is distinct from the submodule's current commit.)

tests/tr1/run.pl
Mention runbe.pl in lowercase, to match the file itself.

# Checklist

Be sure you've read README.md and understand the scope of this repo.

If you're unsure about a box, leave it unchecked. A maintainer will help you.

- [x] Identifiers in product code changes are properly `_Ugly` as per
  https://eel.is/c++draft/lex.name#3.1 or there are no product code changes.
- [ ] The STL builds successfully and all tests have passed (must be manually
  verified by an STL maintainer before automated testing is enabled on GitHub,
  leave this unchecked for initial submission).
- [x] These changes introduce no known ABI breaks (adding members, renaming
  members, adding virtual functions, changing whether a type is an aggregate
  or trivially copyable, etc.).
- [ ] These changes were written from scratch using only this repository,
  the C++ Working Draft (including any cited standards), other WG21 papers
  (excluding reference implementations outside of proposed standard wording),
  and LWG issues as reference material. If they were derived from a project
  that's already listed in NOTICE.txt, that's fine, but please mention it.
  If they were derived from any other project (including Boost and libc++,
  which are not yet listed in NOTICE.txt), you *must* mention it here,
  so we can determine whether the license is compatible and what else needs
  to be done.

### CODE REVIEW

File: tests/std/run.pl [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+use strict;
+use File::Basename;
+
+# This file is just a stub to find and execute the real run.pl which is
```
Comment (BillyONeal) [inline tests/std/run.pl]: This suggests we forgot a file.

Comment (StephanTLavavej) [reply tests/std/run.pl]: It refers to a file on the Microsoft-internal side of things; I can clean up this comment though (as I believe it provides no value).

Comment (BillyONeal) [reply tests/std/run.pl]: Can we copy the real run.pl and run.pm here?

Comment (StephanTLavavej) [reply tests/std/run.pl]: That would be significant Perl surgery which I would prefer not to do. We virtually never need to modify those files when updating tests - the only times I've had to are when we've added new compilers like Clang. Ultimately, the new test runner that @cbezault is working on will be used by contributors to this GitHub repo, and the Perl infrastructure will be a minor curiosity (especially if they're driven by the same `env.lst` files, which are the main thing we need to modify when adding tests). Hopefully we can make the Microsoft-internal test runner Contest use the GitHub test runner and eliminate this Perl dependency (much as we hope to make the Microsoft-internal MSBuild system use the CMake build system and eliminate `stl/msbuild` which is a similar relic).

I considered filing an issue to track this, but I don't think we should perform such surgery at all.

**Pull request merged**